### PR TITLE
[CHORE] Many performance improvements & couple of improvements

### LINKED ISSUE #1951: [BUG] Various problems with latest message in channel overview
Here are a few things I noticed in the latest beta version of the app related to the channel overview.

### 1. Nothing is displayed if the latest message was an emoji

img_5204

img_5206

Sometimes (especially after a re-login) it seems to work, but most of the times nothing is rendered at all.

### 2. Although messages are present, "no message" is displayed

img_5207

img_5208

As you can see (or not thanks to blur), the channel has heaps of messages :)

### 3. Instead of the latest message, the penultimate message is displayed

img_5211

img_5210

I think thats it for now :) Thanks guys for your efforts to make this app even greater!

- Your Rocket.Chat app version: 3.0 beta latest
- Your Rocket.Chat server version: 0.66.3
- Device (or Simulator) you're running with: iPhone 7, iOS 11 latest

### LINKED ISSUE #1952: [BUG] Wrong order of channels in overview
It's just the default channel #general which is sticked to the top of my channel list. Everything else is sorted in the right way.

img_5212

- Your Rocket.Chat app version: 3.0 beta latest
- Your Rocket.Chat server version: 0.66.3
- Device (or Simulator) you're running with: iPhone 7, iOS 11 latest

### PR DESCRIPTION
**Pull request by rafaelks**
@RocketChat/ios

This PR improves the performance of the app by changing parts of the WebSocket message handling, simplifying the reutilisation of the SubscriptionCell and a couple of other small stuff. It also removes unused assets from the project and updates the Default icon asset.

Closes #1951 
Closes #1952

# Progress

- [x] Reduce 100% of the bottlenecks in the Subscriptions list;
- [x] Remove unused assets in the project (not related to performance);
- [x] Improve performance on avatarView reutilisation;
- [x] Start using the new button for "Scroll to Bottom" on the messages list, and improve it's performance;
- [x] Cache the cell height calculation for message cells;
- [x] Improve the performance of the messages list;

# To be done next, in a separated PR

- Reduce the amount of DB calls (Realm) on user status updates. There will be another way to fetch this data from the back-end in the future, but we've this problem right now and we need to deal with it.

### CODE REVIEW

File: Rocket.Chat/Controllers/Chat/ChatViewController.swift [hunk 1]
```diff
@@ -200,10 +219,10 @@ final class ChatViewController: SLKTextViewController {
     }
 
     override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
-        super.viewWillTransition(to: size, with: coordinator)
-
-        coordinator.animate(alongsideTransition: nil, completion: { _ in
-            self.collectionView?.collectionViewLayout.invalidateLayout()
+        dataController.invalidateLayout(for: nil)
```
Comment (filipealva) [inline Rocket.Chat/Controllers/Chat/ChatViewController.swift]: It was not clear that `nil` would invalidate the entire cache until I double checked the implementation of `invalidateLayout(for:)`. Let's improve it.

**Comment by Sameesunkaria** [comment]
It seems like the missing last text message bug is only happening on this branch. I was not able to reproduce it on `develop`.


img_5359

**Comment by cardoso** [comment]
@Sameesunkaria I'm reproducing this issue on develop. After switching servers.

**Comment by rafaelks** [comment]
@cardoso @Sameesunkaria PR already addresses this issue, thanks guys!

**Pull request merged**